### ISSUE TITLE
Params of Pipeline is not found in Records.Data

### ISSUE THREAD

**Issue by vvvsrx** [tektoncd/results#101]
# Expected Behavior
```
"spec": {
    "params": [
        {
            "name": "gitrevision",
            "value": "9de276f1bb9078c967358ec7aa6fb1a8e8a56b29"
        },
        {
            "name": "gitrepositoryurl",
            "value": "git@github.com:tektoncd/results.git"
        },
        {
            "name": "homepage",
            "value": "https://github.com/tektoncd/results.git"
        }
    ],
    "pipelineRef": {
        "name": "pipeline-XXX"
    },
    "serviceAccountName": "robot",
    "timeout": "1h0m0s",
    "workspaces": [
        {
            "name": "sonar-vault",
            "secret": {
                "secretName": "sonar-auth"
            }
        }
    ]
}
```

# Actual Behavior

None

# Steps to Reproduce the Problem

1.
2.
3.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:10:43Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12", GitCommit:"7cd5e9086de8ae25d6a1514d0c87bac67ca4a481", GitTreeState:"clean", BuildDate:"2020-11-12T09:11:15Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
  ```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

   Tekton version : v0.21.0

**Comment by wlynch** [comment]
Thanks for the report!

This looks like we're dropping some fields when storing the object. We should look into generating the spec from the upstream source.

**Comment by wlynch** [comment]
Update on this - I've been looking into upstream proto generation for the pipeline types so that we can use that as the source of truth. I've been able to mostly get this mostly working, but the URL field in knative/pkg statuses is causing issues.

I'm trying to avoid forking these libraries if possible, so I opened https://github.com/knative/pkg/issues/2099 for more discussion.

**Comment by wlynch** [comment]
Following up on old issues - I believe this was fixed with https://github.com/tektoncd/results/pull/121 and should be available in v0.3.x
